### ISSUE TITLE
PublicationYear is not generated from metadata when uploading json file in fabrica

### ISSUE THREAD

**Issue by kjgarza** [datacite/lupo#286]
this seems to be a API problem and can be reproduced with Rspec.

{
  "data": {
    "attributes": {
      "doi": "10.14454/119f96",
      "confirmDoi": null,
      "url": null,
      "creators": [],
      "titles": [],
      "publisher": null,
      "publicationYear": null,
      "descriptions": [],
      "xml": "ewogICJ0eXBlcyI6IHsKICAgICJyZXNvdXJjZVR5cGUiOiAiRGlzc2VydGF0aW9uIiwKICAgICJyZXNvdXJjZVR5cGVHZW5lcmFsIjogIlRleHQiCiAgfSwKICAiY3JlYXRvcnMiOiBbCiAgICB7CiAgICAgICJuYW1lVHlwZSI6ICJQZXJzb25hbCIsCiAgICAgICJnaXZlbk5hbWUiOiAiRGF2aWQiLAogICAgICAiZmFtaWx5TmFtZSI6ICJDdXBvIiwKICAgICAgImFmZmlsaWF0aW9uIjogWwogICAgICAgICJEcmV4ZWwgVW5pdmVyc2l0eSIKICAgICAgXSwKICAgICAgIm5hbWVJZGVudGlmaWVycyI6IFsKICAgICAgICB7CiAgICAgICAgICAic2NoZW1lVXJpIjogImh0dHBzOi8vb3JjaWQub3JnIiwKICAgICAgICAgICJuYW1lSWRlbnRpZmllciI6ICJodHRwczovL29yY2lkLm9yZy8wMDAwLTAwMDEtNTYxNy04NzYyIiwKICAgICAgICAgICJuYW1lSWRlbnRpZmllclNjaGVtZSI6ICJPUkNJRCIKICAgICAgICB9CiAgICAgIF0KICAgIH0KICBdLAogICJ0aXRsZXMiOiBbCiAgICB7CiAgICAgICJsYW5nIjogImVuIiwKICAgICAgInRpdGxlIjogIlRpdGxlIiwKICAgICAgInRpdGxlVHlwZSI6IG51bGwKICAgIH0sCiAgICB7CiAgICAgICJsYW5nIjogImVuIiwKICAgICAgInRpdGxlIjogInN1YnRpdGxlIiwKICAgICAgInRpdGxlVHlwZSI6ICJTdWJ0aXRsZSIKICAgIH0KICBdLAogICJwdWJsaXNoZXIiOiAiRHJleGVsIFVuaXZlcnNpdHkiLAogICJwdWJsaWNhdGlvblllYXIiOiAyMDE5LAogICJkZXNjcmlwdGlvbnMiOiBbCiAgICB7CiAgICAgICJsYW5nIjogImVuIiwKICAgICAgImRlc2NyaXB0aW9uIjogIkFic3RyYWN0eSBnb29kbmVzcyIsCiAgICAgICJkZXNjcmlwdGlvblR5cGUiOiAiQWJzdHJhY3QiCiAgICB9CiAgXSwKICAicHJvdmlkZXJJZCI6ICJzbWwiLAogICJjbGllbnRJZCI6ICJzbWwuZHJleGVsIiwKICAiYWdlbmN5IjogIkRhdGFDaXRlIgp9",
      "source": "fabrica",
      "state": "draft",
      "reason": null,
      "event": null,
      "mode": "upload"
    },
    "type": "dois"
  }
}

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_l4xv)

**Comment by richardhallett** [comment]
I spent a fair amount of time tracking this down but it turns out it's not exactly a bug.
To clarify the original problem a bit:

The original user reported sending DataCite JSON (note not JSON:API or Schema.org) and was using the file upload within the Fabrica UI to do this. With this metadata:

```{
    "agency": "DataCite",
    "clientId": "sml.drexel",
    "creators": [
        {
            "affiliation": [
                "Drexel University"
            ],
            "familyName": "Cupo",
            "givenName": "David",
            "nameIdentifiers": [
                {
                    "nameIdentifier": "https://orcid.org/0000-0001-5617-8762",
                    "nameIdentifierScheme": "ORCID",
                    "schemeUri": "https://orcid.org"
                }
            ],
            "nameType": "Personal"
        }
    ],
    "descriptions": [
        {
            "description": "Abstracty goodness",
            "descriptionType": "Abstract",
            "lang": "en"
        }
    ],
    "providerId": "sml",
    "publicationYear": 2019,
    "publisher": "Drexel University",
    "titles": [
        {
            "lang": "en",
            "title": "Title"
        },
        {
            "lang": "en",
            "title": "subtitle",
            "titleType": "Subtitle"
        }
    ],
    "types": {
        "resourceType": "Dissertation",
        "resourceTypeGeneral": "Text"
    }    
}```

If you run this through bolognese you will get a publicationYear of nil, which is why it is then not set in other places e.g. Fabrica.

The problem comes because the above JSON is being treated as the format "crosscite" which has a slightly different JSON format, namely it's snake_case for attribute names not camelCase) and so the publicationYear or anything else that is not single word or snake_case will just not get parsed.
It appears to "work" with just missing data because some of the file format can be parsed.

This only happens when you have a scenario where you send through without any information about the format you're sending, which is the case if you just send raw JSON without any file type information (i.e. base64 in JSON API xml attribute)

Bolognese attempts to parse this file and look for something to identify it.

To make it work just add "schemaVersion": "http://datacite.org/schema/kernel-4" to the above json and bolognese will parse correctly as format of "datacite_json"

In summary there isn't really a fix for bolognese here, it's more that the API doesn't have a way of specifying a type in the base64 blob and that's not easy to change without a API upgrade.

I'm closing as there isn't a bug it's more just the way it works, specify a schemaVersion and the json will be parsed okay.

**Comment by mfenner** [comment]
I would suggest we do some work in bolognese to either correctly detect the format, or fail with an explicit error. This would be in https://github.com/datacite/bolognese/blob/master/lib/bolognese/utils.rb#L395-L417 and should be a separate issue.
